paris/src/main/java/com/airbnb/paris/proxies/ViewProxy.kt

```diff
@@ -82,10 +84,25 @@ class ViewProxy(view: View) : BaseProxy<ViewProxy, View>(view) {
             if (margin != null) {
                 marginParams.setMargins(margin, margin, margin, margin)
             } else {
-                marginBottom?.let { marginParams.bottomMargin = it }
-                marginLeft?.let { marginParams.leftMargin = it }
-                marginRight?.let { marginParams.rightMargin = it }
-                marginTop?.let { marginParams.topMargin = it }
+                if (marginHorizontal?.let { it >= 0 } == true) {
+                    marginHorizontal?.let {
+                        marginParams.leftMargin = it
+                        marginParams.rightMargin = it
+                    }
+                } else {
+                    marginLeft?.let { marginParams.leftMargin = it }
+                    marginRight?.let { marginParams.rightMargin = it }
+                }
+
+                if (marginVertical?.let { it >= 0 } == true) {
+                    marginVertical?.let {
+                        marginParams.bottomMargin = it
+                        marginParams.topMargin = it
+                    }
+                } else {
+                    marginBottom?.let { marginParams.bottomMargin = it }
+                    marginTop?.let { marginParams.topMargin = it }
+                }
```

Comment:
Could we make it so that there's a single assignment for each side? I think it could improve readability.